### ISSUE TITLE
App get crashed while interacting with scrollview - PanModal.

### ISSUE THREAD

**Issue by parasjoshicasting** [slackhq/PanModal#197]
Hi,

Is anyone facing any crash issues with the method didPanOnScrollView of the class PanModalPresentationController? 
Here are the crash logs with some lines where we can identify the reason.

UIKitCore                      0x1c79a0 -[UIScrollView _setContentOffset:animated:animationCurve:animationAdjustsForContentOffsetDelta:animation:animationConfigurator:] + 824
0x2574ec PanModalPresentationController.didPanOnScrollView(_:change:) + 732 (PanModalPresentationController.swift:732)
0x257350 closure #1 in PanModalPresentationController.observe(scrollView:) + 4299109200 (<compiler-generated>:4299109200)

Kindly let me know if anyone facing the issue or has any idea about that.

Thank you

**Comment by Brandon-T** [comment]
`PanModalPresentationController` calls `didPanOnScrollView` which calls `haltScrolling` which modifies `contentOffset` when it calls `showVerticalScrollIndicator = true` which then in turn, calls the observer which calls `didPanOnScrollView`.

**TLDR:** `didPanOnScrollView -> haltScrolling -> observe(scrollView:) -> didPanOnScrollView -> haltScrolling -> observe(scrollView:) -> .........`

Infinite recursion -> stack overflow -> crash

**Comment by PhanithNY** [comment]
Just got the same crash, my issue is with memory leaks.
Resolved the leak, and issue is gone. 
Hope that helps.
